src/overlays/actors/ovl_player_actor/z_player.c

```diff
@@ -2979,12 +2986,12 @@ s32 func_80835644(PlayState* play, Player* this, Actor* arg2) {
 
 void func_80835688(Player* this, PlayState* play) {
     if (!func_80835644(play, this, this->heldActor)) {
-        Player_SetUpperActionFunc(this, func_808356E8);
+        Player_SetUpperActionFunc(this, Player_UpperAction_CarryActor);
         LinkAnimation_PlayLoop(play, &this->upperSkelAnime, &gPlayerAnim_link_normal_carryB_wait);
     }
 }
 
-s32 func_808356E8(Player* this, PlayState* play) {
+s32 Player_UpperAction_CarryActor(Player* this, PlayState* play) {
```

Comment:
it seems (looking at `func_8083A0F4` which does `Player_SetupAction(Player_Action_80846050) > func_80835688 > Player_SetUpperActionFunc(Player_UpperAction_CarryActor)`) this is indeed the main/default path for carrying stuff,
but `func_8083A0F4` also seems to have a few hardcoded things where the player "carries" but doesn't use `Player_UpperAction_CarryActor` (notably silver rocks? `ACTOR_EN_ISHI` large)

I guess we're sweeping those exceptions under the rug for now

Reply:
the special cases are all full action functions, as in his whole body is involved. The upper action is only for his upper body and you can still, for example, walk around

Reply:
to be more clear, something like the silver rock or the gold gauntlets block are "too heavy" and use their own action function where you are stuck in place while holding the actor, while the stuff that use `Player_UpperAction_CarryActor` let you still move around and such